src/main/java/org/kiwiproject/dropwizard/util/startup/StartupLocker.java

```diff
@@ -0,0 +1,151 @@
+            var curatorFramework = curatorFrameworkHelper.startCuratorFramework(curatorConfig);
+            var lock = curatorLockHelper.createInterProcessMutex(curatorFramework, lockPath);
+
+            try {
+                tryAcquireStartupLock(lock, lockPath, lockTimeout);
+                environment.lifecycle().addLifeCycleListener(new StartupWithLockJettyLifeCycleListener(curatorFramework, lock, lockPath, executioner));
+                return StartupLockInfo.builder()
+                        .client(curatorFramework)
+                        .lock(lock)
+                        .lockPath(lockPath)
+                        .lockState(StartupLockInfo.LockState.ACQUIRED)
+                        .infoMessage("Lock acquired")
+                        .build();
+            } catch (LockAcquisitionException e) {
+                LOG.warn("Lock on path [{}] not obtained. Closing Curator.", lockPath);
+                curatorFrameworkHelper.closeQuietly(curatorFramework);
+
+                return StartupLockInfo.builder()
+                        .lockState(StartupLockInfo.LockState.ACQUIRE_FAIL)
+                        .infoMessage("Failed to obtain startup lock")
+                        .exception(e)
+                        .build();
+            }
+        }
+
+        return StartupLockInfo.builder()
+                .lockState(StartupLockInfo.LockState.NOT_ATTEMPTED)
+                .infoMessage(format("No ZooKeepers are available from connect string [{}]", curatorConfig.getZkConnectString()))
+                .build();
+    }
+
+    private void tryAcquireStartupLock(InterProcessMutex lock, String lockPath, Duration lockTimeout) {
+        LOG.debug("Start lock acquisition for path [{}]. Timeout set to [{}]", lockPath, lockTimeout);
+        curatorLockHelper.acquire(lock, lockTimeout.getQuantity(), lockTimeout.getUnit());
+        LOG.debug("Acquired lock on path [{}]", lockPath);
+    }
+
+    /**
+     * Adds a {@link StartupJettyLifeCycleListener} in case the lock was never acquired.
+     *
+     * @param lockInfo    the lock info indicating if the lock was acquired
+     * @param environment the Dropwizard environment used to add the listener
+     */
+    public void addFallbackJettyStartupLifeCycleListener(StartupLockInfo lockInfo, Environment environment) {
+        if (lockInfo.getLockState() != StartupLockInfo.LockState.ACQUIRED) {
+            environment.lifecycle().addLifeCycleListener(new StartupJettyLifeCycleListener(executioner));
+        }
+    }
+
+    /**
+     * Cleans up the startup lock if it was acquired.
+     *
+     * @param lockInfo the lock info indicating if the lock was acquired
+     */
+    public void releaseStartupLockIfPresent(StartupLockInfo lockInfo) {
+        if (lockInfo.getLockState() == StartupLockInfo.LockState.ACQUIRED) {
+            LOG.warn("Due to exception caught running app [{}], early releasing lock [{}] on path [{}]", this, lockInfo.lock, lockInfo.lockPath);
```

Comment:
This message implies `this` is the "running app" which is not true. In addition, it is also saying there was some exception caught even though the lock state is `ACQUIRED` which doesn't make sense. I think this message needs to be changed, and it should not be at WARN level.